test/e2e/resize_nodes.go

```diff
@@ -0,0 +1,331 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"bytes"
+	"fmt"
+	"os/exec"
+	"strconv"
+	"time"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/wait"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+func resizeNodeInstanceGroup(size int) error {
+	// TODO: make this hit the compute API directly instread of shelling out to gcloud.
+	output, err := exec.Command("gcloud", "preview", "managed-instance-groups", "--zone="+testContext.CloudConfig.Zone,
+		"resize", testContext.CloudConfig.NodeInstanceGroup, fmt.Sprintf("--new-size=%v", size)).CombinedOutput()
+	if err != nil {
+		Logf("Failed to resize node instance group: %v", string(output))
+	}
+	return err
+}
+
+func waitForNodeInstanceGroupSize(size int) error {
+	for start := time.Now(); time.Since(start) < 4*time.Minute; time.Sleep(20 * time.Second) {
+		// TODO: make this hit the compute API directly instread of shelling out to gcloud.
+		output, err := exec.Command("gcloud", "preview", "managed-instance-groups", "--zone="+testContext.CloudConfig.Zone,
+			"describe", testContext.CloudConfig.NodeInstanceGroup).CombinedOutput()
+		if err != nil {
+			Logf("Failed to describe node instance group: %v", string(output))
+			continue
+		}
+		pattern := "currentSize: "
+		i := bytes.Index(output, []byte(pattern))
```

Comment:
I wish would use the API directly and access the objects, as doing string parsing here is gross...

But given it'd probably be best to just get this in, can you please clean up the size getting by doing the following:

- refactor it into a helper function (e.g. `func nodeInstanceGroupSize() (int, error)`)
- use the `strings` golang package instead of `bytes` (`json` would be nicest but we don't want to define the whole returned type, and MIG json output from gcloud is currently weirdly duplicated)
- use golang style if/err blocks rather than elses. For example:

  ```go
func nodeInstanceGroupSize() (int, error) {
  // get "output" through the gcloud call
  pattern := "currentSize: "
  i := strings.Index(pattern)
  if i  == -1 {
    return -1, fmt.Errof("could not find '%s' in the output '%s'", pattern, output)
  }
  // ...
}
  ```

  While it's more intuitive to check affirmatively in an `if` block, golang style is to check negatively (whether an error happened) in an `if` block, and proceed through the function only upon success. This keeps indentation small.